tools/dependencies.py

```diff
@@ -0,0 +1,183 @@
+                        tmp, directories = line.split(maxsplit=2)
+                        if directories != "-":
+                            self.privileged = True
+                    else:
+                        match = Forest.TEXT.match(line)
+                        if match and match.group("text"):
+                            self.text = match.group("text")
+                        elif Forest.CATALOG.match(line):
+                            self.user_visible = True
+
+            forest.nodes[path.name] = self
+
+        def add_child(self, path, forest):
+            if path.name in forest.nodes:
+                child = forest.nodes[path.name]
+                self.children.add(child)
+                forest.roots.discard(child)
+            else:
+                self.children.add(forest.Node(path, forest))
+
+        @property
+        def name(self):
+            return self.path.name
+
+        @property
+        def color(self):
+            return "lightgray" if self.privileged else "white"
+
+        @property
+        def style(self):
+            return "filled" if self.user_visible else "filled,dashed"
+
+        def __repr__(self):
+            return (f"Node({self.text}, name={self.name}, "
+                    f"#children={len(self.children)}, "
+                    f"privileged={self.privileged}, "
+                    f"user_visible={self.user_visible})")
+
+        def __iter__(self):
+            yield self
+            for child in self.children:
+                yield from child
+
+    def __init__(self):
+        self.nodes = dict()
+        self.roots = set()
+
+    @classmethod
+    def build(cls, input_directory):
+        forest = cls()
+        for path in input_directory.iterdir():
+            if not path.name.endswith(".permission") or not path.is_file():
+                continue
+            if path.name not in forest.nodes:
+                forest.add_root_node_from_path(path)
+        return forest
+
+    def add_root_node_from_path(self, path):
+        self.roots.add(self.Node(path, self))
+
+    def digraph(self):
+        dot = graphviz.Digraph()
+        for node in self.nodes.values():
+            dot.attr('node', style=node.style, fillcolor=node.color)
+            dot.node(node.name, node.text)
+            for child in node.children:
+                dot.edge(node.name, child.name, arrowhead='vee')
+        return dot
+
+    def __repr__(self):
+        return f"Forest(#nodes={len(self.nodes)}, #roots={len(self.roots)})"
+
+    def __iter__(self):
+        for root in self.roots:
+            yield from root
+
+def visualize(output_file, input_directory, view, debug):
+    forest = Forest.build(input_directory)
+    if debug:
+        print(forest)
+        for node in forest:
+            print(node)
+    digraph = forest.digraph()
+    digraph.attr(pack='true')
+    unflatten = digraph.unflatten(stagger=2)
+    if output_file is None:
+        with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
This shouldn't be here. I forgot to remove it after changing this to use mktemp().